eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -41,35 +41,85 @@ final class OyRemote implements Objectionary {
     /**
      * The address template.
      */
-    private final String template;
+    private final UrlOy template;
 
     /**
      * Constructor.
      * @param hash Commit hash
      * @throws IOException if fails.
      */
     OyRemote(final CommitHash hash) throws IOException {
-        this.template = String.format(
-            "https://raw.githubusercontent.com/objectionary/home/%s/objects/%%s.eo",
+        this.template = new UrlOy(
+            "https://raw.githubusercontent.com/objectionary/home/%s/objects/%s.eo",
             hash.value()
         );
     }
 
     @Override
     public String toString() {
-        return this.template;
+        return this.template.toString();
     }
 
     @Override
     public Input get(final String name) throws MalformedURLException {
-        final URL url = new URL(
-            String.format(this.template, name.replace(".", "/"))
-        );
+        final URL url = this.template.value(name);
         Logger.debug(
             this, "The object '%s' will be pulled from %s...",
             name, url
         );
         return new InputOf(url);
     }
 
+    /**
+     * Objectionary URL template.
+     *
+     * @since 1.0
+     */
+    static final class UrlOy {
+
+        /**
+         * URL template. Expects two placeholders in terms of
+         * {@link String#format(String, Object...)}: 1st for hash,
+         * 2nd for program name.
+         * <br/>Example: "https://raw.githubusercontent.com/objectionary/home/%s/objects/%s.eo"
+         */
+        private final String template;
+
+        /**
+         * Objects version hash.
+         */
+        private final String hash;
+
+        /**
+         * Ctor.
+         * @param template URL template.
+         * @param hash Version hash.
```

Comment:
@mximp please, add more context for `Version` in `JavaDoc`- it confuses if you see the code first time

Reply:
added

Reply:
@volodya-lombrozo I think there is too much info exposed here.. In the end any string can be used and it's not necessarily git hash.

Reply:
@volodya-lombrozo In general case it can be any identifier of objects snapshot/version. For `UrlOy` it's important to stress out that it has objects versioning. It can be files on a file share stored by version.

Reply:
Also I believe what you suggest is explained in `CommitHash` and `ChRemote`

Reply:
@mximp ok, but I don't see the problem to have verbose javadocs. It only helps to understand the idea. When I was reading the code first time - it wasn't obvious what is `hash` where it went from.

Reply:
@volodya-lombrozo Concise javadocs are easer to process. Maybe the notion `hash` is not the best option for this object.